### PR TITLE
RFR: Merge results class

### PR DESCRIPTION
**Pull request by aamontal**
The results of merges now get wrapped in more explicit classes, resolved and conflicted. The former output was a hash with a type key. 

Also includes changes to the output type, which used to be an array with positional values, is now a merge_result class that has info about success, conflict etc.

### CODE REVIEW

File: lib/dyph3/support/collater.rb [hunk 1]
```diff
@@ -2,40 +2,26 @@ module Dyph3
...
+      private
         # @param [in] conflicts
         # @returns the list of conflicts with contiguous parts merged if they are non_conflicts
         def merge_non_conflicts(res, i = 0)
           while i < res.length - 1
-            if res[i][:type] == :non_conflict && res[i+1][:type] == :non_conflict
-              #res[i][:text] += "\n" unless res[i][:text][-1] == "\n"
-              res[i][:text] += res[i+1][:text]
+            if res[i].resolved? && res[i+1].resolved?
+              res[i].combine(res[i+1])
```
Comment (jelder) [inline lib/dyph3/support/collater.rb]: This method looks like it might be simpler as a `reduce`.

File: spec/lib/dyph3/differ_spec.rb [hunk 2]
```diff
@@ -10,9 +10,9 @@
           left =  "ants bears cat dog".split
           base =  "ants bears cat dog".split
           right =  "ants elephant cat bears dog".split
-          result = Dyph3::Differ.merge_text(left, base, right, split_function: identity , join_function: identity, current_differ: current_differ)
-          expect(result[0]).to eq right
-          expect(result[1]).to be false
+          merge_result = Dyph3::Differ.merge_text(left, base, right, split_function: identity , join_function: identity, current_differ: current_differ)
+          expect(merge_result.joined_results).to eq right
+          expect(merge_result.success?).to be true
```
Comment (jelder) [inline spec/lib/dyph3/differ_spec.rb]: I prefer to write expectations like this as:

```ruby
expect(merge_result).to be_success
```

File: lib/dyph3/support/collater.rb (hunk 1 above)
Comment (aamontal) [inline lib/dyph3/support/collater.rb]: Oh yes, was planning to do that, but got distracted. Will refactor that

File: spec/lib/dyph3/differ_spec.rb (hunk 2 above)
Comment (aamontal) [inline spec/lib/dyph3/differ_spec.rb]: @jelder  When I get time to refactor the original_heckell_differ class I will refactor the specs a bit.

**Pull request merged**